src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -0,0 +1,335 @@
+    public CommandBase singulatorPossesion() {
+        return new CommandBase() {
+            {
+                addRequirements(Indexer.this);
+            }
+
+
+            @Override
+            public boolean isFinished() {
+                return frontIntakeIR.getAsBoolean();
+                
+                //frontIntakeIR.getVoltage() > irSensorVoltage;
+                // values of .8 when open, 1.7 when closed
+
+            }
+
+            @Override
+            public void execute() {
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                singulator.set(0);
+            }
+
+        };
+
+    }
+    //This command will make sure that the singulator has possesion of the ball
+
+    public CommandBase pierrePossesion() {
+        return new CommandBase() {
+
+                @Override
+                public void initialize() {
+                if (frontIntakeIR.getAsBoolean()) {
+                    
+                    singulator.set(singulatorMotorSpeed);
+                }
+                if (backIntakeIR.getAsBoolean()) {
+                    pierreMotor.set(pierreIndexSpeed);
+                    singulator.set(singulatorMotorSpeed);
+                }
+            }
+             
+
+            public boolean isFinished() {
+                return bottomPierreIR.getAsBoolean();
+                // values of .8 when open, 2.4 when closed
+
+            }
+
+            @Override
+            public void execute() {
+                if (frontIntakeIR.getAsBoolean()) {
+                  
+                    singulator.set(singulatorMotorSpeed);
+                }
+                if (backIntakeIR.getAsBoolean()) {
+                    pierreMotor.set(pierreIndexSpeed);
+                    singulator.set(singulatorMotorSpeed);
+                }
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                pierreMotor.set(0);
+                singulator.set(0);
+                SmartDashboard.putNumber("Ball Count", ballCounting);
+
+            }
+
+        };
+
+    }
+    // This command will make sure that pierre has possesion of the ball. It will be at the bottom
+
+    public CommandBase loadBallToTop() {
+        return new CommandBase() {
+
+            {
+                addRequirements(Indexer.this);
+            }
+
+            @Override
+            public boolean isFinished() {
+                return topPierreIR.getAsBoolean();
+                // values of .8 when open, 2.4 when closed
+
+            }
+
+            @Override
+            public void execute() {
+                pierreMotor.set(pierreIndexSpeed);
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                pierreMotor.set(0);
+               
+            }
+
+        };
+
+    }
+    //this will bring the ball to the top
+    
+    public CommandBase unLoadBall() {
```

Comment:
We should mention that a ball must already be at the top of Pierre for this command to work.